python/tank/platform/import_stack.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Manages a thread-safe stack of Toolkit imports.
+"""
+
+import threading
+
+
+class ImportStack(object):
+    """
+    Manages a thread-safe stack of Toolkit imports.
+    """
+
+    # global variable that holds a stack of references to
+    # a current bundle object, per thread - this variable is populated
+    # whenever the bundle.import_module method is executed
+    # and is a way for import_framework() to be able to resolve
+    # the current bundle even when being recursively called
+    # inside an import_module call.
```

Comment:
I would add something like:

By using a threadlocal storage, we effectively maintain one stack per thread. This allows for threadsafe execution and imports, where two pieces of different code can execute in parallel and resolve the correct imports.